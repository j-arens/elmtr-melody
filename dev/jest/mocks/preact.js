module.exports = {
    unmountComponent: jest.fn(),
};
