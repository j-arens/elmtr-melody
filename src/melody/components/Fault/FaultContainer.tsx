import BaseButton from '@components/BaseButton/';
import { h } from 'preact';
const s = require('./style.scss');

interface Props {
    children?: JSX.Element[];
    allowReload?: boolean;
}

export default ({ children, allowReload = true }: Props) => (
    <div class={s.errorContainer}>
        {children}
        {allowReload &&
            <BaseButton
                onClick={window.location.reload.bind(window.location)}
                className={s.errorContainer__reloadBtn}
            >
                Reload
            </BaseButton>
        }
    </div>
);
