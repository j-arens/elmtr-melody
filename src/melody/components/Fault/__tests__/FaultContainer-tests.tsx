import { render } from 'preact-render-to-string';
import { h } from 'preact';
import FaultContainer from '../FaultContainer';

describe('FaultContainer', () => {
    it('renders', () => {
        const component = render(
            <FaultContainer>
                <div>alpine white</div>
            </FaultContainer>
        );
        expect(component).toMatchSnapshot();
    });

    it('renders without a reload button', () => {
        const component = render(
            <FaultContainer allowReload={false}>
                <div>alpine white</div>
            </FaultContainer>
        );
        expect(component).toMatchSnapshot();
    });
});
