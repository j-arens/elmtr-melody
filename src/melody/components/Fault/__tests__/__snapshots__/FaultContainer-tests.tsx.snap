// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaultContainer renders 1`] = `
<div class="errorContainer">
  <div>alpine white</div>
  <button class="btn errorContainer__reloadBtn">Reload</button>
</div>
`;

exports[`FaultContainer renders without a reload button 1`] = `
<div class="errorContainer">
  <div>alpine white</div>
</div>
`;
